File: examples/node/01_basic_evaluate.ts
/**
 * Basic evaluate demo using the AGR TypeScript SDK.
 *
 * Usage:
 *   npm install agr-sdk  (or: pnpm add agr-sdk)
 *   AGR_API_KEY=agr_sk_... npx ts-node 01_basic_evaluate.ts
 */

const BASE_URL = process.env.AGR_BASE_URL ?? "http://localhost:8000";
const API_KEY = process.env.AGR_API_KEY ?? "";

if (!API_KEY) {
  console.error("ERROR: Set AGR_API_KEY environment variable");
  process.exit(1);
}

const headers = {
  Authorization: `Bearer ${API_KEY}`,
  "Content-Type": "application/json",
};

interface EvaluateRequest {
  agent_id: string;
  action: string;
  resource: string;
  context?: Record<string, unknown>;
  approver_email?: string;
}

interface EvaluateResponse {
  decision: string;
  reason: string;
  policy_id: string | null;
  approval_id: string | null;
  latency_ms: number;
  eval_id: string;
  risk_score: number | null;
  risk_level: string | null;
  compliance_findings: Array<{
    plugin: string;
    standard: string;
    rule_id: string;
    severity: string;
    message: string;
    passed: boolean;
  }> | null;
}

async function evaluate(req: EvaluateRequest): Promise<EvaluateResponse> {
  const resp = await fetch(`${BASE_URL}/v1/evaluate`, {
    method: "POST",
    headers,
    body: JSON.stringify(req),
  });
  if (!resp.ok) {
    throw new Error(`HTTP ${resp.status}: ${await resp.text()}`);
  }
  return resp.json() as Promise<EvaluateResponse>;
}

const testCases: Array<{ label: string; req: EvaluateRequest }> = [
  {
    label: "Low-risk read (expect ALLOW)",
    req: { agent_id: "trusted-reader", action: "read", resource: "docs.txt" },
  },
  {
    label: "Production db.drop (expect DENY)",
    req: {
      agent_id: "coder-001",
      action: "db.drop",
      resource: "prod-db",
      context: { environment: "production" },
    },
  },
  {
    label: "Large bulk delete (risk engine may upgrade)",
    req: {
      agent_id: "agent-9999",
      action: "delete",
      resource: "user-table",
      context: { count: 500000, environment: "production" },
    },
  },
];

(async () => {
  for (const { label, req } of testCases) {
    const result = await evaluate(req);
    const risk = result.risk_level ?? "n/a";
    console.log(`[${result.decision.padEnd(20)}] risk=${risk.padEnd(6)} ${label}`);
  }
})();

File: examples/python/01_basic_evaluate.py
"""Basic evaluate example using the AGR Python SDK.

Usage:
    pip install agr-sdk
    AGR_API_KEY=agr_sk_... python 01_basic_evaluate.py
"""

import os
import sys

sys.path.insert(0, "../../packages/agr-core")

import httpx

AGR_BASE_URL = os.environ.get("AGR_BASE_URL", "http://localhost:8000")
AGR_API_KEY = os.environ.get("AGR_API_KEY", "")

if not AGR_API_KEY:
    print("ERROR: Set AGR_API_KEY environment variable")
    sys.exit(1)

headers = {"Authorization": f"Bearer {AGR_API_KEY}"}

cases = [
    {
        "label": "Low-risk read (expect ALLOW)",
        "payload": {
            "agent_id": "trusted-reader",
            "action": "read",
            "resource": "docs/readme.txt",
            "context": {},
        },
    },
    {
        "label": "Production db.drop (expect DENY)",
        "payload": {
            "agent_id": "coder-001",
            "action": "db.drop",
            "resource": "prod-db",
            "context": {"environment": "production"},
        },
    },
    {
        "label": "Staging deploy (expect ALLOW)",
        "payload": {
            "agent_id": "deploy-bot",
            "action": "deploy",
            "resource": "staging-server",
            "context": {"environment": "staging"},
        },
    },
]

with httpx.Client(base_url=AGR_BASE_URL, headers=headers) as client:
    for case in cases:
        resp = client.post("/v1/evaluate", json=case["payload"])
        data = resp.json()
        decision = data.get("decision", "ERROR")
        risk = data.get("risk_level", "n/a")
        print(f"[{decision:20s}] risk={risk:6s}  {case['label']}")
        if data.get("compliance_findings"):
            violations = [f for f in data["compliance_findings"] if not f["passed"]]
            if violations:
                for v in violations:
                    print(f"  ⚠  [{v['standard']}/{v['rule_id']}] {v['message']}")
